SimPEG/EM/Base.py

```diff
@@ -202,16 +204,26 @@ def MfMui(self):
             self._MfMui = self.mesh.getFaceInnerProduct(self.mui)
         return self._MfMui
 
-    def MfMuiDeriv(self, u):
+    def MfMuiDeriv(self, u, v=None, adjoint=False):
         """
         Derivative of :code:`MfMui` with respect to the model.
         """
         if self.muiMap is None:
             return Utils.Zero()
 
-        return (
-            self.mesh.getFaceInnerProductDeriv(self.mui)(u) * self.muiDeriv
-        )
+        if getattr(self, '_MfMuiDeriv', None) is None:
+            self._MfMuiDeriv = self.mesh.getFaceInnerProductDeriv(
```

Comment:
instead of creating a separate property, I stash it as a private variable here

Reply:
I am not sure this is doing the same thing. My main purpose was to update them for the inversion. 
For instance in second iteration I need to update `_MfMuiDeriv`, but this property is not None, so my 
guess update does not happen with this way.

Reply:
thanks for catching this, I will add MfMuiDeriv to the `_clear_on_mu_update` list above